arcgis-ios-sdk-samples/Utility Network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -0,0 +1,261 @@
+            // A nil category will let the trace fail.
+            self.setStatus(message: "Category not set for filter barrier.")
+            return nil
+        }
+        // Note: AGSUtilityNetworkAttributeComparison or AGSUtilityCategoryComparison with AGSUtilityCategoryComparisonOperator.doesNotExist
+        // can also be used. These conditions can be joined with either AGSUtilityTraceOrCondition or AGSUtilityTraceAndCondition.
+        let comparison = AGSUtilityCategoryComparison(category: category, comparisonOperator: .exists)
+        configuration.filter?.barriers = comparison
+        configuration.includeIsolatedFeatures = isolationSwitch.isOn
+        
+        let parameters = AGSUtilityTraceParameters(traceType: .isolation, startingLocations: [startingLocationElement])
+        parameters.traceConfiguration = configuration
+        return parameters
+    }
+    
+    func loadUtilityNetwork() {
+        setStatus(message: "Loading Utility Network…")
+        // Load the utility network to be ready to run a trace against it.
+        utilityNetwork.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.setStatus(message: "Loading Utility Network failed.")
+                self.presentAlert(error: error)
+            } else {
+                let networkDefinition = self.utilityNetwork.definition
+                let domainNetwork = networkDefinition.domainNetwork(withDomainNetworkName: self.domainNetworkName)
+                let utilityTier = domainNetwork?.tier(withName: self.tierName)
+                self.filterBarrierCategories = networkDefinition.categories
+                // Get the trace configuration for the specified utility tier.
+                self.traceConfiguration = utilityTier?.traceConfiguration
+                // Create a trace filter.
+                self.traceConfiguration?.filter = AGSUtilityTraceFilter()
+                
+                // Get a default starting location.
+                let networkSource = networkDefinition.networkSource(withName: self.networkSourceName)
+                let assetGroup = networkSource?.assetGroup(withName: self.assetGroupName)
+                let assetType = assetGroup?.assetType(withName: self.assetTypeName)
+                if let type = assetType, let element = self.utilityNetwork.createElement(with: type, globalID: self.globalId) {
+                    self.startingLocationElement = element
+                } else {
+                    self.setStatus(message: "Creating starting location failed.")
+                    return
+                }
+                // Draw the starting location element on the map.
+                self.drawStartingLocation()
+                self.setStatus(message: "Utility Network loaded.")
+                self.setUIState()
+            }
+        }
+    }
+    
+    // MARK: UI and Feedback
+    
+    func drawStartingLocation() {
```

Comment:
This function is to avoid the 40-line lint constraint for a single function, as well as to separate UI updates from utility network related methods.